File: p9/main.cpp
#include <iostream>
using namespace std;

class A {
    private:
    public:
    // constructor:
    A() {}

    void m1() {
        cout << "AAAA" << endl;
    }

    void m() {
        this->m1();
    }
};

class B : public A {
    private:
    public:
    B() {}
    
    void m1() {
        cout << "BBBB" << endl;
    }
};

int main(void) {
    B b;
    b.m(); // muestra AAAA
    return 0;
}

File: p9/Main.java
class A {

    public A() {}

    public void m1() {
        System.out.println("AAAA");
    }

    public void m() {
        this.m1();
    }
}

class B extends A {
        
    public B() {}

    public void m1() {
        System.out.println("BBBB");
    }
}

public class Main {
    public static void main(String[] args) {
        B b = new B();
        b.m();
    }
}
